compiler/rustc_ty_utils/src/opaque_types.rs

```diff
@@ -56,10 +55,9 @@ impl<'tcx> OpaqueTypeCollector<'tcx> {
 }
 
 impl<'tcx> TypeVisitor<TyCtxt<'tcx>> for OpaqueTypeCollector<'tcx> {
-    type BreakTy = ErrorGuaranteed;
-
     #[instrument(skip(self), ret, level = "trace")]
-    fn visit_ty(&mut self, t: Ty<'tcx>) -> ControlFlow<ErrorGuaranteed> {
+    fn visit_ty(&mut self, t: Ty<'tcx>) -> ControlFlow<!> {
+        t.super_visit_with(self)?;
```

Comment:
Can you explain to me what the material change is in this commit ("Stop failing eagerly, and collect all opaque types even if some are erroneous")? 

I don't see any tests associated with it, so is it to just suppress spurious errors or what?